OpenRA.Game/Graphics/Theater.cs

```diff
@@ -68,6 +68,9 @@ public Theater(TileSet tileset)
 					var indices = t.Value.Frames != null ? t.Value.Frames : Enumerable.Range(0, frameCount);
 					variants.Add(indices.Select(j =>
 					{
+						if (j >= allFrames.Length)
+							throw new InvalidDataException(i + " defined more frames than the sprite has.");
```

Comment:
It might make more sense to not just mention the image name here, but also the template id imo (if we can). I tested this error with the D2k tileset, which re-uses the same image with different frame definitions for all templates (which I assume is the common usecase), so just the image name is not that helpful. `BLOXBASE.R8 defined more frames than the sprite has.`

Reply:
Good idea. Done.